x-pack/plugins/reporting/public/components/report_error_button.tsx

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiButtonIcon, EuiCallOut, EuiLoadingSpinner, EuiPopover } from '@elastic/eui';
+import React, { Component } from 'react';
+import { jobQueueClient } from '../lib/job_queue_client';
+
+interface Props {
+  jobId: string;
+}
+
+interface State {
+  isLoading: boolean;
+  isPopoverOpen: boolean;
+  calloutTitle: string;
+  error?: string;
+}
+
+export class ReportErrorButton extends Component<Props, State> {
+  private mounted?: boolean;
+
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      isLoading: false,
+      isPopoverOpen: false,
+      calloutTitle: 'Unable to generate report',
+    };
+  }
+
+  public render() {
+    return (
+      <EuiPopover
+        id="popover"
+        button={this.renderButton()}
+        isOpen={this.state.isPopoverOpen && !this.state.isLoading}
+        closePopover={this.closePopover}
+      >
+        <EuiCallOut color="danger" title={this.state.calloutTitle}>
+          <p>{this.state.error}</p>
+        </EuiCallOut>
+      </EuiPopover>
+    );
+  }
+
+  public componentWillUnmount() {
+    this.mounted = false;
+  }
+
+  public componentDidMount() {
+    this.mounted = true;
+  }
+
+  private renderButton = () => {
+    if (this.state.isLoading) {
+      return <EuiLoadingSpinner onClick={this.onButtonClick} size="m" />;
+    }
+
+    return (
+      <EuiButtonIcon
+        onClick={this.onButtonClick}
+        iconType="alert"
+        color={'danger'}
+        aria-label="Show report error"
+      />
+    );
+  };
+
+  private onButtonClick = () => {
+    this.setState({
+      isPopoverOpen: !this.state.isPopoverOpen,
+    });
+
+    if (!this.state.error) {
+      this.loadError();
+    }
+  };
+
+  private closePopover = () => {
+    this.setState({
+      isPopoverOpen: false,
+    });
+  };
+
+  private loadError = async () => {
+    this.setState({ isLoading: true });
+    let reportContent;
```

Comment:
This is `any` as well.  As much as possible we want to avoid `any` since you lose all the benefits of typings when you do that.  For instance, once you type this out, typescript will warn you that you might be referencing an undefined object below.

<img width="756" alt="screen shot 2018-09-18 at 8 34 44 am" src="https://user-images.githubusercontent.com/16563603/45687834-c6e2d400-bb1d-11e8-8e95-b65b70fb80c3.png">
<img width="720" alt="screen shot 2018-09-18 at 8 33 15 am" src="https://user-images.githubusercontent.com/16563603/45687835-c6e2d400-bb1d-11e8-8185-8ca97f2be4d4.png">